Numbered list template

duarte opened the pull request:
This adds support for rendering the ‘Numbered List’ display hint

In numbered list articles:
• Headlines get a new ’splash headline’ style
• H2 headings get a number next to them
• there is a new ‘H3’ heading level
• review stars get special styling
• ‘thumbnail’ size images are right aligned and circular/oval

This includes dark mode support

**Plus:** also removes the pre-garnett Titlepiece font, which is no longer used and was easily confused with the new Titlepiece font.

***

<h3>Header (before)</h3>

![_Users_duarte_carrilho_Code_mobile-apps-article-templates_fixture_view-from-nhs-staff-in-frontline-of-battle-against-coronavirus html (2)](https://user-images.githubusercontent.com/4561/80373714-c54ec580-8895-11ea-9768-7df9ee7b627e.png)


<h3>Header (after)</h3>

![_Users_duarte_carrilho_Code_mobile-apps-article-templates_fixture_view-from-nhs-staff-in-frontline-of-battle-against-coronavirus html (3)](https://user-images.githubusercontent.com/4561/80373737-d1d31e00-8895-11ea-8a91-cb89973318cc.png)

***

<h3>Header, on tablet (before)</h3>

> ![_Users_duarte_carrilho_Code_mobile-apps-article-templates_fixture_view-from-nhs-staff-in-frontline-of-battle-against-coronavirus html (1)](https://user-images.githubusercontent.com/4561/80373808-ea433880-8895-11ea-81f8-c2de6c98a146.png)


<h3>Header, on tablet (after)</h3>

> ![_Users_duarte_carrilho_Code_mobile-apps-article-templates_fixture_view-from-nhs-staff-in-frontline-of-battle-against-coronavirus html](https://user-images.githubusercontent.com/4561/80373820-ee6f5600-8895-11ea-97ec-dfde9cf8a613.png)

***

<h3>Numbered headings (before)</h3>

> ![_Users_duarte_carrilho_Code_mobile-apps-article-templates_fixture_life-after-coronavirus-panic-buying-cocaine-drug-dealer-spaceman-therapist html (2)](https://user-images.githubusercontent.com/4561/80373876-08109d80-8896-11ea-957b-52084653585a.png)

</h3>Numbered headings (after)</h3>

> ![_Users_duarte_carrilho_Code_mobile-apps-article-templates_fixture_life-after-coronavirus-panic-buying-cocaine-drug-dealer-spaceman-therapist** html](https://user-images.githubusercontent.com/4561/80373886-0d6de800-8896-11ea-96c1-f3bd8312ed1e.png)

webb04 commented:
Test articles: https://www.theguardian.com/world/2020/mar/31/view-from-nhs-staff-in-frontline-of-battle-against-coronavirus & https://www.theguardian.com/world/2020/mar/28/life-after-coronavirus-panic-buying-cocaine-drug-dealer-spaceman-therapist

webb04 commented:
It's also on dotcom, but should this man be in a circle?

![Uploading Screen Shot 2020-04-27 at 17.12.43.png…]()

webb04 commented:
Design looks good, especially on Dark mode!

Spotted some things which I'll post some screenshots.

webb04 commented:
The right floated image seems to clash with adverts (seen from column3)

<img width="526" alt="Screen Shot 2020-04-27 at 17 26 21" src="https://user-images.githubusercontent.com/11618797/80396852-351c7a80-88ad-11ea-9582-d725606701ac.png">

webb04 commented:
I can see the series on column3 but there's no red background on column4 and it doesn't show on column1 and column2, is that intentional? also is it supposed to link to the series page?

<img width="776" alt="Screen Shot 2020-04-27 at 17 25 47" src="https://user-images.githubusercontent.com/11618797/80396951-5b421a80-88ad-11ea-8444-eb157fb827e5.png">

webb04 commented:
I can't remember if @benwuersching added a different shade of red for dark mode?

| White square on panic buying cocaine article dark mode | dark mode red colour |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/11618797/80397178-a9efb480-88ad-11ea-825f-3294ce86baed.png" width="300" /> | <img src="" width="https://user-images.githubusercontent.com/11618797/80397238-bf64de80-88ad-11ea-888a-9f1e58932402.png" /> |

benwuersching commented:
Good morning @webb04, are you referring to the red block containing the H1? We should use news-200 in this case. You can use our dark mode designs for liveblogs as a guideline for all pillars: https://app.zeplin.io/project/58ee27a3e08030c8234f88a3/dashboard?seid=5d6d12dd5af8a69be0d40b23
Refer to the colour block containing the liveblog H1 for each pillar.

duarte commented:
Hey, I have adopted the news-200 colour to the splash header (and equivalent colours in other pillars), and worked out the spacing issues with the ad slot.

Also fixed the issue with the tag (it was rendering differently in Chrome and Safari!). The behaviour with the series tag linking, but the section tag not linking isn't changed — it's something that was discussed way back, and I think has actually now changed on dotcom, but out of scope here.

I agree about the oval thumbnail images. It's best in square crops. But I don't want to crop non-square images, as it might be intended behaviour from editorial, and it seems they like to use it. So for now, let's leave that bit as is

duarte commented:
> White square on panic buying cocaine article dark mode	dark mode red colour
> <img alt="" width="300" src="https://user-images.githubusercontent.com/11618797/80397178-a9efb480-88ad-11ea-825f-3294ce86baed.png">	<img alt="" width="300" src="https://user-images.githubusercontent.com/11618797/80397238-bf64de80-88ad-11ea-888a-9f1e58932402.png">

I forgot to add, about this white square — that's an embedded atom, so I think not in scope for the template here! :)

webb04 commented:
Good stuff, I'll take another look and through the code 👍

I'll make a ticket for us to fix that embedded atom style (if it's empty)

webb04 commented on ArticleTemplates/assets/js/modules/numberedList.js:
```diff
@@ -0,0 +1,59 @@
+// import { getElementOffset, debounce } from 'modules/util';
```
Can this be removed?

webb04 commented on ArticleTemplates/assets/scss/modules/content/_standfirst.scss:
```diff
@@ -17,6 +17,8 @@
     font-family: $egyptian-display;
     font-weight: 400;
+    width: calc(100% - #{$gs-unit*6});
+    max-width: $gs-unit*76;
```
Maths 🧮

webb04 commented on webpack.config.js:
```diff
@@ -17,102 +18,104 @@ module.exports = (env, argv) => {
+      ]
+    },
+    plugins: [
+      new LiveReloadPlugin(),
```
I wonder what this does for the production build? Do you think the live reload and fixtures are worth keeping? considering we'll hopefully be closing this repo soon

duarte replied:
I didn't realise the livereload would affect prod! Just in case I'll remove it or keep that in a branch so I can use it for dev-only, as there are a couple more things I'm hoping to do on this repo! Fixtures are definitely ok to delete, I think :)

webb04 commented:
Also I wonder if we can remove adverts on numberedLists (Might send a note to product)

<img width="402" alt="Screen Shot 2020-04-29 at 17 57 30" src="https://user-images.githubusercontent.com/11618797/80628090-c116d980-8a48-11ea-8ab8-69647aafc758.png">

webb04 commented:
Spotted on iPad pro landscape, white square

![Simulator Screen Shot - iPad Pro (12 9-inch) (3rd generation) - 2020-04-30 at 09 43 18](https://user-images.githubusercontent.com/11618797/80690915-683a5600-8ac7-11ea-9468-cfe5080459f4.png)

duarte commented:
Thanks very much for your reviews Jamie! :)

benwuersching commented:
@duarte Looking forward to seeing this on the app 😊!
